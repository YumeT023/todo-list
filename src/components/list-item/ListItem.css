.container__list {
  padding: 0.3rem;
  max-height: 27.8rem;
  overflow-y: scroll;
}

.list-item--full {
  max-height: 32.5rem;
}