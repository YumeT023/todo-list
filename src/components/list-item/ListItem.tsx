import React, { useCallback } from "react";
import { Item } from "../item";
import { TodoItem } from "../../models/Todo.model";
import "./ListItem.css";

export type ListItemProps = {
  items: Array<TodoItem>;
  whoseComplete: boolean;
  handleCheck(idx: number): void;
};

export const ListItem = ({
  items,
  whoseComplete,
  handleCheck,
}: ListItemProps) => {
  const renderItems = useCallback(() => {
    return items.map(
      (item) =>
        item.isComplete === whoseComplete && (
          <Item
            key={item.idx}
            item={item}
            onChange={() => handleCheck(item.idx)}
          />
        )
    );
  }, [items]);

  return <div className="container__list">{renderItems()}</div>;
};
