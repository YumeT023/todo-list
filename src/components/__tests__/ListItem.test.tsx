import { create } from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import { ListItem } from "../list-item";
import { itemsMock } from "./mocks/items-mock";

describe("ListItem", () => {
  describe("rendering", () => {
    test("rendered correctly", () => {
      const listItem = render(<ListItem items={[]} handleCheck={() => null} />);

      expect(listItem.baseElement).toBeInTheDocument();
    });

    test("[snapshot] rendered correctly", () => {
      const listItem = create(<ListItem items={[]} handleCheck={() => null} />);

      const json = listItem.toJSON();
      expect(json).toMatchSnapshot();
    });
  });

  describe("Items mapping", () => {
    it("map items according to `whoseComplete` props", () => {
      render(
        <ListItem
          items={itemsMock}
          handleCheck={() => null}
          whoseComplete={false}
        />
      );

      const listItemUnfinished = screen.getByTestId("list-unfinished");
      const unfinishedItemLength = itemsMock.filter(
        (item) => !item.isComplete
      ).length;

      expect(listItemUnfinished.children).toHaveLength(unfinishedItemLength);
    });

    it("[snapshot] map items according to `whoseComplete` props", () => {
      const renderer = create(
        <ListItem
          items={itemsMock}
          handleCheck={() => null}
          whoseComplete={false}
        />
      );

      const json = renderer.toJSON();
      expect(json).toMatchSnapshot();
    });
  });
});
