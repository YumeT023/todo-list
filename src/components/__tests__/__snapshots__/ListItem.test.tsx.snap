// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem Items mapping [snapshot] map items according to \`whoseComplete\` props 1`] = `
<div
  className="container__list "
  data-testid="list-unfinished"
>
  <div
    className="item item--unfinished"
    data-testid="item--unfinished"
  >
    <div
      className="item__content"
    >
      content__1
    </div>
    <div
      className="item__action"
    >
      <input
        name="item-status"
        onChange={[Function]}
        type="checkbox"
      />
    </div>
  </div>
  <div
    className="item item--unfinished"
    data-testid="item--unfinished"
  >
    <div
      className="item__content"
    >
      content__3
    </div>
    <div
      className="item__action"
    >
      <input
        name="item-status"
        onChange={[Function]}
        type="checkbox"
      />
    </div>
  </div>
</div>
`;

exports[`ListItem rendering [snapshot] rendered correctly 1`] = `
<div
  className="container__list "
  data-testid="list-done"
/>
`;
